### PR TITLE
Adding fixes for warnings and necissary imports

### PR DESCRIPTION
**Pull request by arrouse**
@Raizlabs/maintainers-ios @nbonatsakis this expands a bit more on your Pull request #112  that removes warnings for iOS8 as well as adds imports for all the files as we move away from PCH

### CODE REVIEW

**Comment by arrouse** [comment]
@Raizlabs/maintainers-ios Gonna merge this in at the end of the day unless anyone has any comments.

File: RZUtils/Components/RZSingleChildContainerViewController/RZSingleChildContainerViewController.m [hunk 1]
```diff
@@ -297,14 +297,27 @@ - (void)completeTransition:(BOOL)didComplete
 
 - (UIViewController *)viewControllerForKey:(NSString *)key
 {
+    UIViewController *ret = nil;
```
Comment (KingOfBrian) [inline RZUtils/Components/RZSingleChildContainerViewController/RZSingleChildContainerViewController.m]: not a fan of `ret` -- Rename 'viewController' ?

File: RZUtils/Components/RZSingleChildContainerViewController/RZSingleChildContainerViewController.m [hunk 2]
```diff
@@ -297,14 +297,27 @@ - (void)completeTransition:(BOOL)didComplete
...
-        return self.toVC;
+        ret = self.toVC;
     }
     
-    return nil;
+    return ret;
+}
+- (UIView *)viewForKey:(NSString *)key
+{
+    UIView *ret = nil;
```
Comment (KingOfBrian) [inline RZUtils/Components/RZSingleChildContainerViewController/RZSingleChildContainerViewController.m]: Same as above

File: RZUtils/Components/RZLocationService/RZLocationService.m [hunk 3]
```diff
@@ -121,14 +121,14 @@ - (void)locationWithCompletion:(RZLocationServiceSuccessBlock)successBlock
...
-    return ([CLLocationManager authorizationStatus] == kCLAuthorizationStatusAuthorized || [CLLocationManager authorizationStatus] == kCLAuthorizationStatusNotDetermined)
+    return ( ([CLLocationManager authorizationStatus] == kCLAuthorizationStatusAuthorizedAlways || [CLLocationManager authorizationStatus] == kCLAuthorizationStatusAuthorizedWhenInUse) || [CLLocationManager authorizationStatus] == kCLAuthorizationStatusNotDetermined)
             && [CLLocationManager locationServicesEnabled];
 }
 
 - (void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status
 {
     // if the status is Authorized and we were waiting for authorization, start the timeout timer. 
-    if (status == kCLAuthorizationStatusAuthorized && self.fetchLocInProgress)
+    if ((status == kCLAuthorizationStatusAuthorizedAlways || status == kCLAuthorizationStatusAuthorizedWhenInUse) && self.fetchLocInProgress)
```
Comment (KingOfBrian) [inline RZUtils/Components/RZLocationService/RZLocationService.m]: High Value Comment:   I think we're missing a space.

**Comment by arrouse** [comment]
Thanks for the comments @KingOfBrian are you ok with me merging this?

**Comment by arrouse** [comment]
Per our convo: merging.

**Pull request merged**